yarn-project/end-to-end/src/fixtures/utils.ts

```diff
@@ -264,15 +276,20 @@ export async function setup(numberOfAccounts = 1, opts: SetupOptions = {}): Prom
 
   const logger = getLogger();
   const hdAccount = mnemonicToAccount(MNEMONIC);
+  const privKeyRaw = hdAccount.getHdKey().privateKey;
+  const publisherPrivKey = privKeyRaw === null ? null : Buffer.from(privKeyRaw);
 
   if (PXE_URL) {
     // we are setting up against a remote environment, l1 contracts are assumed to already be deployed
     return await setupWithRemoteEnvironment(hdAccount, config, logger, numberOfAccounts);
   }
 
-  const deployL1ContractsValues = await setupL1Contracts(config.rpcUrl, hdAccount, logger);
-  const privKeyRaw = hdAccount.getHdKey().privateKey;
-  const publisherPrivKey = privKeyRaw === null ? null : Buffer.from(privKeyRaw);
+  let deployL1ContractsValues: DeployL1Contracts;
+  if (!opts.deployL1ContractsValues) {
+    deployL1ContractsValues = await setupL1Contracts(config.rpcUrl, hdAccount, logger);
+  } else {
+    deployL1ContractsValues = opts.deployL1ContractsValues;
+  }
```

Comment:
```suggestion
  const deployL1ContractsValues = opts.deployL1ContractsValues ?? await setupL1Contracts(config.rpcUrl, hdAccount, logger);
```

I know, it's just a style change, but I hate using `let` when `const` will do.